utils.py

```diff
@@ -6673,8 +6673,42 @@ def isGoodForNERSC(self, no_step=False):
         #good = good &
         return good
 
+    def getTimeInfoForChain(self):
+        timeInfo = dict()
+        if 'Chain' in self.request['RequestType']:
+            base = self.request['RequestType'].replace('Chain','')
+            itask=1
+            carry_on = {}
+            while True:
+                t = '%s%d'%(base, itask)
+                itask+=1
+                if t in self.request:
+                    task = self.request[t]
+                    timeInfo[t] = {
+                        'tpe':task.get('TimePerEvent')/task.get('FilterEfficiency',1),
+                        'cores':task.get('Multicore',1)
+                    }
+                else:
+                    break
+        return timeInfo
+
     def isGoodToConvertToStepChain(self ,keywords=None, talk=False):
         all_same_arch = True
+
+        ## efficiency 
+        time_info = self.getTimeInfoForChain()
+        totalTimePerEvent = 0
+        efficiency = 0
+        max_ncores = 1
+        for i,info in time_info.items():
+            totalTimePerEvent += info['tpe']
+            efficiency += info['tpe']*info['cores']
+            if info['cores']>max_ncores: max_ncores = info['cores']
+        # print "Total time per event for TaskChain: %0.1f" % totalTimePerEvent
+        efficiency /= totalTimePerEvent*max_ncores
+        # print "CPU efficiency of StepChain with %u cores: %0.1f%%" % (max_ncores,efficiency*100)
+        acceptable_efficiency = efficiency>0.75
```

Comment:
We could also set this 0.75 to be configurable in https://github.com/CMSCompOps/WmAgentScripts/blob/master/unifiedConfiguration.json
E.g. adding an entry in unified config
```
 "efficiency_threshold_for_stepchain" : {
   "value" : 0.75,
   "description" : "Minimum efficiency threshold for step chain conversion"
 },
```
and in this `utils.py` we do
`acceptable_efficiency = efficiency>UC.get("efficiency_threshold_for_stepchain")`
So that one year later whoever maintaining Unified can just change the config file without tracing back to the codebase.

Reply:
done